### ISSUE TITLE
UpdateNames & Util.AvgMaxOutput - wrong oldPower at 3+ panels

### ISSUE THREAD

**Issue by kennumen** [mezzodrinker/spas#12]
rev9 UpdateNames on line 450 and same for line 478 Utilities.AverageMaxOutput
```
	void UpdateNames(out float oldPower) {
		oldPower = 0.0f;
		for (int i = 0; i < solarPanels.Count; i++) {

<snip>

			if (oldPower == 0) oldPower = panelOldOutput;
			else oldPower = (oldPower + panelOldOutput) / 2;
// (4 + 0) / 2 = 2 -> OK
// ( [ (10 + 8) / 2] + 0) / 2 = 4.5f -> NOK real average is 6
		}
	}
```

instead:
...
	void UpdateNames(out float oldPower) {
		oldPower = 0.0f;

		int i = 0;
		for (; i < solarPanels.Count; i++) {
			IMySolarPanel panel = solarPanels[i];

			float panelOutput = Utilities.MaxOutput(panel);
			string[] split = panel.CustomName.Split('~');
			if (split.Length <= 1) {
				panel.SetCustomName(panel.CustomName + "~" + panelOutput);
				continue;
			}

			float panelOldOutput;
			float.TryParse(split[split.Length - 1], out panelOldOutput);

			string name = "";
			for (int k = 0; k < split.Length - 1; k++) name += split[k];
			name += "~" + panelOutput;
			panel.SetCustomName(name);

			oldPower += panelOldOutput;
		}
		oldPower = oldPower / (i + 1);
	}
...

and instead:
...
		public static float AverageMaxOutput(List<IMySolarPanel> panels) {
			float power = 0.0f;
			int i = 0;
			for (; i < panels.Count; i++) {
				IMySolarPanel panel = panels[i];

				power += MaxOutput(panel);
			}
			power = power / (i + 1);
			return power;
		}
...

**Comment by mezzodrinker** [comment]
You're right, that seems to have been a major brain fart. Thanks for showing me.

Also, it should probably be `power /= i` instead of `power /= i + 1` because at the end of the loop, `i` always equals `panels.Count`

**Comment by kennumen** [comment]
Well now we've both had a brainfart.

Not 100% how C# handles division by 0 but you'll probably need to check `i != 0` in that case.

**Comment by mezzodrinker** [comment]
AFAIK, division by zero results in `float.NaN`, but I could be mistaken. There's a `DivisionByZeroException` as well.

Anyway, that can't happen in the specific case being the output methods because their arguments are never empty. At least not in the code that I wrote.
